keps/sig-auth/3766-referential-authorization/README.md

```diff
@@ -659,8 +759,46 @@ moving ReferenceGrant to the new API group, just notes to save discussion time.
 * Clarify that the expected operating model for implementations expects them to
   have broad, read access to both ReferenceGrant and the specific `To` Kinds they
   support, and then self-limit to only _use_ the relevant ones.
+* Determine if strings used in `For` are completely separate from verbs.
+* Decide if supporting verbs such as `[create, update, delete]` make sense.
+* Decide if supporting virtual verbs such as `[forward, scale, dance]` make sense.
+  We already have `For` which has a similar function.
+* Decide whether to expose/hide the API based on the API server feature flag
+  and/or authorizer config.
+* Decide how cross-namespace ReferenceGrants join target resources to their
+  originating reference. One way of accomplishing this is to extract the
+  namespace alongside the name of a target reference when reconciling 
+  ClusterReferenceGrants, then pair those with a matching ReferenceGrant. A 
+  trade-off of this approach is that it makes ReferenceGrants
+  dependent on ClusterReferenceGrants. The alternative is that the subject of a ClusterReferenceConsumer is authorized for the granted object regardless of 
+  whether there is actually a cross-namespace reference.
 * Decide whether to formally add `*` as a special value for Namespace, to mean
   "all namespaces".
+* Decide whether to formally add `*` as a special value for classNames, to mean
+  "all classNames". Should access to a class-partitioned API be open or closed 
+  for a ClusterReferenceConsumer that lists no classNames? Software authors, API
+  owners, and cluster admins all own different pieces of the config here. An 
+  optional field that fails open can easily be missed by a cluster admin. An  
+  opt-in wildcard can become canonical in practice.
```

Comment:
but what about APIs without the Class concept?

Reply:
I think Rob means he's convinced `classNames` are required for Consumers to receive authorization, only when API Authors have indicated that there is a `classPath` on their cluster-level Grants.
Basically if `classNames` is missing, it will fail closed as opposed to failing open and granting access.

In the case where the API is not split on `classPaths`, no `classNames` is required for any access.